serchProc.js

```diff
@@ -0,0 +1,122 @@
+let storeName, storeNumber, roadName, menu, storeX, storeY, id, writer;
+let num = 1;
+
+//모든 리스트를 보여주는 함수
+function getData() {
+    console.log('getData!!!!!!')
+    const xhr = new XMLHttpRequest();
+    xhr.open('GET', '/api/storeDB', true);
+    xhr.onreadystatechange = function () { // Call a function when the state changes.
+        if (this.readyState === XMLHttpRequest.DONE && this.status === 200) {
+            // Request finished. Do processing here.
+            const responseData = JSON.parse(this.response);
+            console.log(responseData);
+            //list를 출력하는 for
+            for (var i = 0; i <= responseData.length; i++) {
+                storeName = responseData[i].storeName;
+                storeNumber = responseData[i].storeNumber;
+                id = responseData[i]._id;
+                roadName = responseData[i].roadName;
+                storeX = responseData[i].storeX;
+                storeY = responseData[i].storeY;
+                writer = responseData[i].writer;
+                menu = responseData[i].menu;
+
+                const button = `<button id='btn' onclick='showData("${id}", "${storeName}", "${storeNumber}", "${roadName}", "${storeX}")'>정보보기</button>`;
+                document.getElementById("resultList").innerHTML += num + "." + storeName + storeNumber+ button + "<br>";
+                //document.getElementById("resultList").innerHTML += button + "<br>";
+                num++;
+            }
+        }
+    }
+    xhr.send(null);
+}
+getData()
+
+//db를 검색하는 함수
+function serchDb() {
+    const type = document.getElementById('serchType').value;
+    const sKey = document.getElementById('serching').value;
+
+    var xhr = new XMLHttpRequest();
+    xhr.open('GET', "/api/storeDB/serchKey?serchId=" + sKey + "&type=" + type, true);
+    xhr.onreadystatechange = function () { // Call a function when the state changes.
+        console.log('this.readyState: ', this.readyState);
+        console.log('this.state:', this.state);
+
+        if (this.readyState === XMLHttpRequest.DONE && this.status === 200) {
+            document.getElementById('resultList').innerHTML = "";
+
+            // Request finished. Do processing here.
+            const responseList = JSON.parse(this.response);
+            console.log(responseList.length);
+            console.log(responseList);
+            //clearDiv();
+            if (responseList.length == 0) {
```

Comment:
```suggestion
            if (responseList.length === 0) {
```
자바스크립트에서는 비교연산시 일반적으로 `===` or `!==`을 사용하는것이 좋습니다.
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Comparison_Operators

Reply:
수정하였습니다!